service/storage_proxy.cc

```diff
@@ -4033,14 +4058,20 @@ void storage_proxy::send_to_live_endpoints(storage_proxy::response_id_type respo
     }
 }
 
-// returns number of hints stored
-template<typename Range>
-size_t storage_proxy::hint_to_dead_endpoints(std::unique_ptr<mutation_holder>& mh, const Range& targets, db::write_type type, tracing::trace_state_ptr tr_state) noexcept
+// Returns the number of hints that have been stored.
+template <typename Range>
+size_t storage_proxy::hint_to_dead_endpoints(std::unique_ptr<mutation_holder>& mh, const Range& targets,
+        db::write_type type, tracing::trace_state_ptr tr_state) noexcept
 {
+    static_assert(std::same_as<std::ranges::range_value_t<Range>, gms::inet_address>, "The range must store IPs");
+
     if (hints_enabled(type)) {
         db::hints::manager& hints_manager = hints_manager_for(type);
-        return boost::count_if(targets, [&mh, tr_state = std::move(tr_state), &hints_manager] (gms::inet_address target) mutable -> bool {
-            return mh->store_hint(hints_manager, target, tr_state);
+        const auto tmptr = get_token_metadata_ptr();
```

Comment:
Same as in other places, we should use captured token_metadata from erm. In this place we can just remove this change if we switch `per_destination_mutation::_mutations` to `host_id`-s.

Reply:
OK